static/js/filter.js

```diff
@@ -496,13 +496,22 @@ export class Filter {
         // stream, stream + topic,
         // is: private, pm-with:,
         // is: mentioned, is: resolved
+        const term_types = this.sorted_term_types();
+        // check for pm-with:"" filter (because it would result in empty search results)
+        if (
+            _.isEqual(term_types, ["pm-with"]) &&
+            this.operands("pm-with").length === 1 &&
+            this.operands("pm-with")[0] === ""
+        ) {
+            return false;
+        }
+
```

Comment:
Can we move this statement to be _before_ the comment that talks about `can_mark_messages_read`. Having the new conditional be between those two is confusing.

Reply:
oh okay, I'll make these changes by tomorrow!